### DISCUSSION TITLE
Single access token for feed authentication?

### DISCUSSION THREAD

**Discussion by ledermann** [jeremyevans/rodauth#333] [General]
In Authlogic, there is a concept of "single access token", useful for feeds:
https://www.rubydoc.info/gems/authlogic/1.3.2#label-Single+access+token

> This token is used for single access only, it is not persisted. Meaning the user provides it, Authlogic grants them access, and that’s it. If they want access again they need to provide the token again. Authlogic will NEVER store this value in the session or a cookie. For added security, by default this token is ONLY allowed for RSS and ATOM requests. Also, this token does NOT change with the password. Meaning if the user changes their password, this token will remain the same. Lastly, this token uses a “friendly” toke (see the URL example below) so that it is easier to email / copy and paste. [...] This is great for private feed access. So your URL to that user’s private feed could look something like:
> http://www.mydomain.com/account/feed.rss?single_access_token=4LiXF7FiGUppIPubBPey

Is there something similar in Rodauth?

**Comment by jeremyevans** [comment]
I don't think there is anything similar in Rodauth currently, but it is unlikely to be a difficult feature to add.  I wouldn't currently be in favor of shipping such a feature in Rodauth (RSS/Atom feed support is no longer popular, and private RSS/Atom feed support seems quite niche). However, I could reconsider depending on community support.

**Comment by ledermann** [comment]
Regarding RSS/Atom feeds, I agree. Here is another argument that might help you reconsider :-)

My use case involves iCalendar support in an application: A user should be able to add time-related events as a subscription to their calendar. The subscription would be a URL to an ICS file, which needs to be secured by a token.

By the way, thanks for this great project! After years of using Devise (and even earlier Authlogic) I am currently evaluating Rodauth for future projects.

**Comment by muellerj** [comment]
Please correct me if I'm wrong, but this sounds pretty similar to what I described in https://github.com/jeremyevans/rodauth/discussions/325? I at least had a similar use case in mind.

I haven't had much time to work on it, but there's a rough first version at https://github.com/muellerj/rodauth-pat.
